cpp/include/raft/neighbors/nn_descent_types.hpp

```diff
@@ -56,6 +56,8 @@ struct index_params : ann::index_params {
   size_t max_iterations            = 20;      // Number of nn-descent iterations.
   float termination_threshold      = 0.0001;  // Termination threshold of nn-descent.
   bool return_distances            = false;   // return distances if true
+  size_t n_clusters                = 2;       // number of clusters if using batching
```

Comment:
We should explain to the user very clearly exaclty what this option means. A user new to RAFT (or cuVS) is not going to understand what this does, nor the implications of the clustering. I would suggest maybe writing up a paragraph or two here about what this option really does. Write out the steps of the algorithm and the benefits/drawbacks for different number of clusters. Why would a user use this option?